src/geode/geosciences_io/mesh/fem_output.cpp

```diff
@@ -0,0 +1,383 @@
+/*
+ * Copyright (c) 2019 - 2024 Geode-solutions
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ */
+#include <fstream>
+#include <memory>
+#include <string>
+#include <vector>
+
+#include <geode/basic/attribute.hpp>
+#include <geode/basic/attribute_manager.hpp>
+#include <geode/basic/io.hpp>
+#include <geode/basic/types.hpp>
+
+#include <geode/basic/filename.hpp>
+
+#include <geode/geosciences_io/mesh/internal/fem_output.hpp>
+
+#include <geode/geometry/point.hpp>
+
+namespace
+{
+    class SolidFemOutputImpl
+    {
+    public:
+        static constexpr geode::index_t OFFSET_START{ 1 };
+        static constexpr char EOL{ '\n' };
+        static constexpr char SPACE{ ' ' };
+        SolidFemOutputImpl(
+            std::string_view filename, const geode::TetrahedralSolid3D& solid )
+            : file_{ geode::to_string( filename ) }, solid_( solid )
+        {
+            OPENGEODE_EXCEPTION(
+                file_.good(), "Error while opening file: ", filename );
+        }
+
+        void write_file()
+        {
+            geode::Logger::info( "[SolidFemOutput::write] Writing fem file." );
+            const auto directories =
+                geode::filepath_without_filename( "filename" );
+            if( !directories.empty() )
+            {
+                std::filesystem::create_directories( directories );
+            }
+            write_problem();
+            write_class();
+            write_dimension();
+            write_scale();
+            write_var_node();
+            write_node_coordinates();
+            write_ref_nodal_dist();
+            write_ref_element_dist();
+            write_nodal_sets();
+            write_element_sets();
+            write_gravity();
+            write_end();
+        }
+
+        void write_problem()
+        {
+            file_ << "PROBLEM:" << EOL;
+        }
+
+        void write_class()
+        {
+            file_ << "CLASS (v.7.202.18152)" << EOL;
+
```

Comment:
restrict the number of spaces

Reply:
Dans les fichiers .fem, j'ai vui qu'il y en avait 4 ici, je peux en mettre moins?